### PR TITLE
eigenexa: add 2.7–2.12

### PR DESCRIPTION
**Pull request by e-kwsm**

### CODE REVIEW

File: var/spack/repos/builtin/packages/eigenexa/package.py [hunk 1]
```diff
@@ -10,9 +10,14 @@ class Eigenexa(AutotoolsPackage):
...
     homepage = "https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/"
-    url = "https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/EigenExa-2.6.tgz"
 
     version("2.6", sha256="a1a4e571a8051443f28e7ea4889272993452a4babd036d2b4dd6b28154302f95")
+    version("2.7", sha256="490f3d0217a8c101e66e785229baaba5b4d674508bc9a5aca6cc5fa074f3a8aa")
+    version("2.8", sha256="3ee846d4db10336d393738eadab2f1c941dfc8fb501f2a4baf0823f0ff938f56")
+    version("2.9", sha256="8788922035bf67abf1a7aecf8e30dd7564de387fda4ecd11c6b4cf9259d25990")
+    version("2.10", sha256="5b1806e132b191d23680b34fbc286d676ba20f58ee754122087a3ec3cacb8fa3")
+    version("2.11", sha256="87dee8ac13f410a007e82df2688fa7f143883229dac729fd20836f4a28fac43d")
+    version("2.12", sha256="2a33999b09d4434a5ce2fbd18cabbfee1cff0b2a12df7ded1f67127157b08f86")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/eigenexa/package.py]: Please reverse the order of these as we prefer newer versions to appear first.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the sha256 and 'spack spec' succeeds.  Please reverse the order of the version directives.

File: var/spack/repos/builtin/packages/eigenexa/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/eigenexa/package.py]: You can see the requested pattern in the first example in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#versions-and-fetching.

Comment (e-kwsm) [reply var/spack/repos/builtin/packages/eigenexa/package.py]: Fixed.

File: var/spack/repos/builtin/packages/eigenexa/package.py [hunk 2]
```diff
@@ -10,8 +10,13 @@ class Eigenexa(AutotoolsPackage):
...
     homepage = "https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/"
-    url = "https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/EigenExa-2.6.tgz"
 
+    version("2.12", sha256="2a33999b09d4434a5ce2fbd18cabbfee1cff0b2a12df7ded1f67127157b08f86")
+    version("2.11", sha256="87dee8ac13f410a007e82df2688fa7f143883229dac729fd20836f4a28fac43d")
+    version("2.10", sha256="5b1806e132b191d23680b34fbc286d676ba20f58ee754122087a3ec3cacb8fa3")
+    version("2.9", sha256="8788922035bf67abf1a7aecf8e30dd7564de387fda4ecd11c6b4cf9259d25990")
+    version("2.8", sha256="3ee846d4db10336d393738eadab2f1c941dfc8fb501f2a4baf0823f0ff938f56")
+    version("2.7", sha256="490f3d0217a8c101e66e785229baaba5b4d674508bc9a5aca6cc5fa074f3a8aa")
     version("2.6", sha256="a1a4e571a8051443f28e7ea4889272993452a4babd036d2b4dd6b28154302f95")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/eigenexa/package.py]: Missed this.  Please add the `url` property you removed from line 13 as an argument to this directive so the sha256 for `2.6` will still work.  The change, with style fixes per `black`, is:

```suggestion
    version(
        "2.6",
        sha256="a1a4e571a8051443f28e7ea4889272993452a4babd036d2b4dd6b28154302f95",
        url="https://www.r-ccs.riken.jp/labs/lpnctrt/projects/eigenexa/EigenExa-2.6.tgz",
    )
```

And I have confirmed that this yields the correct sha256 for that version.

**Review by tldahlgren — DISMISSED** [review]
Thank you for the changes.  I missed one version directive that needs to be fixed.

File: var/spack/repos/builtin/packages/eigenexa/package.py (hunk 2 above)
Comment (e-kwsm) [reply var/spack/repos/builtin/packages/eigenexa/package.py]: I forgot to mention, but `url` is added. Would you review the PR?

**Review by alalazo — APPROVED** [review]

**Pull request merged**